src/main/java/com/templateproject/api/controller/CollectionController.java

```diff
@@ -43,17 +45,16 @@ public CollectionController(CollectionRepository collectionRepository) {
             updatedCollection.setColor(collection.getColor());
             updatedCollection.setFavorite(collection.isFavorite());
             return collectionRepository.save(updatedCollection);
-        } else {
-            return collectionRepository.save(collection);
         }
+        throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Collection non trouvée");
     }
 
-    @DeleteMapping("/collections/{id}")
+    @DeleteMapping("/{id}")
     public @ResponseBody void deleteCollection(@PathVariable(value = "id") Long id) {
         collectionRepository.deleteById(id);
     }
 
-    @PostMapping("/collections/search")
+    @PostMapping("/search")
```

Comment:
For future you could optimise the controller by removing this route and add an optional quey to your "/" routes in order to get allDatas containing the request body part